### ISSUE TITLE
Expose function for tracking page load APIs

### ISSUE THREAD

**Issue by siddharthCommvault** [microsoft/ApplicationInsights-JS#1965]
Hi, 
We at Commvault, need to **monitor the total time taken by all the APIs that gets loaded when a page is loaded** by the user. 

We would like you to expose a function where we can distinguish the page load APIs from the other APIs that are loaded when user click events or performs any other function on the page.
One way of doing this would be to add a new property { 'onPageLoad' : boolean } to the envelope with baseType : 'RemoteDependencyData' and set it true for the apis that are being sent with the first track call.

-We considered updating the page load apis and sending property in response but, this was not feasible and also not at all scalable.
-We tried adding 'onPageload' to custom dimensions for the dependency data but, could not distinguish the page load APIs from others.
-We tried using worker-thread to monitor the APIs and using events updating the flag. But, this is uncertain and we cannot rely on service-workers.

**We have implemented AI using scripts in our app.

**Comment by Karlie-777** [comment]
@siddharthCommvault the functions  [addDependencyListener](https://github.com/microsoft/ApplicationInsights-JS/blob/a98051cf1aa1b32a5a446b749498d77ccc13c212/AISKU/src/Initialization.ts#L739) and [addDependencyInitializer](https://github.com/microsoft/ApplicationInsights-JS/blob/a98051cf1aa1b32a5a446b749498d77ccc13c212/AISKU/src/Initialization.ts#L752) may help to filter and alter API calls as well.

**Comment by siddharthCommvault** [comment]
@Karlie-777 thanks for your response, using these two function we area able to track and update all the dependencies that are being loaded.

We were able to add custom dimensions that helps us to identify the page url of these dependencies.

what we want though is to distinguish those dependencies that are **loaded on a page load** from others that are **loaded on various other events by the user** on the same page. 

Say dependencies a, b & c were loaded when user loads a catalog page and dependencies x, y and z were called while user clicked a button or performed any other action on the same catalog page. We saw that a, b & c are correctly accumulated in first two track calls. We want to add a flag onPageLoad = true to a b & c while adding onPageLoad = false for x y & z

It's just that we want to track all the **remoteDependencyData** envelope that gets called on the first and second _track calls_ for each of our pages.

Is there a way to identify how and when a _track_ is being called and if we could add custom properties to all the envelopes inside any of the _track call_ and not to those of other _track calls_ ?
